queries.sql

```diff
@@ -7,4 +7,41 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
+
+BEGIN;
+UPDATE animals 
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name like '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals
+ROLLBACK;
+SELECT * FROM animals
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '01/01/2022';
+SAVEPOINT delete_by_birth;
+UPDATE animals
+SET weight_kg = weight_kg * (-1);
+ROLLBACK TO delete_by_birth;
+UPDATE animals
+SET weight_kg = weight_kg * (-1)
+WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+/* End transaction */
```

Comment:
Please add a query that shows the average number of escape attempts per animal type of those born between 1990 and 2000, It will look like this  :point_down: 

```sql
 SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/31/12';
```